### ISSUE TITLE
[Bug]: `accessibility.snapshot` unexpectedly pruning more elements in recent update

### ISSUE THREAD

**Issue by querkmachine** [puppeteer/puppeteer#14284]
### Minimal, reproducible example

```TypeScript
/** Extracted from a helper file, normally imported. */
async function getAccessibleName(page, $element) {
  return (
    await page.accessibility.snapshot({
      root: $element
    })
  ).name
}

describe('accessible name', () => {
  it('Should set the accessible name of the show/hide section buttons', async () => {
    // For the code of the `render` function, see:
    // https://github.com/alphagov/govuk-frontend/blob/5024297aee7a01a77364b56cb08ec9130ccaf189/shared/helpers/puppeteer.js#L88-L219
    //
    // For the HTML it outputs, see:
    // https://govuk-frontend-review.herokuapp.com/components/accordion/preview
    //
    // The code that defines the `examples` object has been omitted for brevity.
    await render(page, 'accordion', examples.default)

    const $element = await page.$('.govuk-accordion__section-button')

    await expect(getAccessibleName(page, $element)).resolves.toBe(
      'Section A , Show this section'
    )
  })
})
```

### Background

Upon updating from Puppeteer 24.19.0 to 24.22.3, we found that some of our accessibility tests began to unexpectedly fail with the following error:

```
    expect(received).resolves.toBe()

    Received promise rejected instead of resolved
    Rejected to value: [TypeError: Cannot read properties of null (reading 'name')]

      24 |     const $element = await page.$('.govuk-accordion__section-button')
      25 |
    > 26 |     await expect(getAccessibleName(page, $element)).resolves.toBe(
         |           ^
      27 |       'Section A , Show this section'
      28 |     )
      29 |   })
```

After some digging around, we tried setting `interestingOnly` to `false` when using `accessibility.snapshot` and found that our tests began passing again.

This was unexpected, as [the documentation for `accessibility.snapshot`](https://pptr.dev/api/puppeteer.accessibility.snapshot#remarks) remarks that it prunes "nodes that go unused on most platforms and by most screen readers". However, our test is against the `aria-label` attribute of a `<button>` element, which is _definitely_ used by a number of platforms and screen readers.

For context, the `.govuk-accordion__section-button` HTML looks like this at the time the test is ran:

```html
<button type="button" aria-controls="default-example-content-1" class="govuk-accordion__section-button" aria-expanded="false" aria-label="Section A , Show this section">
  […]
</button>
```

Quick links for the code in GOV.UK Frontend:
- [Accordion component's Puppeteer test file](https://github.com/alphagov/govuk-frontend/blob/5024297aee7a01a77364b56cb08ec9130ccaf189/packages/govuk-frontend/src/govuk/components/accordion/accordion.puppeteer.test.js)
- [Puppeteer helper file](https://github.com/alphagov/govuk-frontend/blob/5024297aee7a01a77364b56cb08ec9130ccaf189/shared/helpers/puppeteer.js) — contains `getAccessibleName` and other helper functions

This appears to be OS agnostic, having been reproduced on macOS, Windows, and Ubuntu Linux. 

### Expectation

We would expect that a `<button>` with an `aria-label` not be pruned from the accessible nodes list, as it is a semantically significant node that is widely used by assistive technologies.

If the definition of an 'interesting' node were to be changed, we would expect this to not happen in a minor release version.

### Reality

Some aspect of how an 'interesting' node is defined has unexpectedly changed between Puppeteer versions 24.19.0 and 24.22.3.

We weren't sure if https://github.com/puppeteer/puppeteer/pull/14221 was related to this, but it appears to be the only interim change that touches on the same area. It may also be something else (such as a bump in the Chromium version) has caused this to change.

### Puppeteer configuration file (if used)

https://github.com/alphagov/govuk-frontend/blob/5024297aee7a01a77364b56cb08ec9130ccaf189/puppeteer.config.js

### Puppeteer version

24.22.3

### Node version

22.20.0

### Package manager

npm

### Package manager version

10.9.3

### Operating system

Linux

**Comment by Lightning00Blade** [comment]
Hey,

I can confirm that with the latest version it's not showing up in the snapshot.
Considering other tools like Chrome DevTools Accessibility also show it I think this is a regression.

**Comment by OrKoN** [comment]
@Lightning00Blade could you please turn the repro into a test?

**Comment by Lightning00Blade** [comment]
It seems to be a nesting issue inside the `h2` element,
A MVRE test is, and it returns null, very likely we think h2 is a leaf node now, taking a look.
```ts
it.only('should work with nested button', async () => {
        const {page} = await getTestState();

        await page.setContent(html`
          <main>
            <h2>
              <button>My Button</button>
            </h2>
          </main>
        `);

        using button = (await page.$('button'))!;
        expect(await page.accessibility.snapshot({root: button})).toMatchObject(
          {
            role: 'button',
            name: 'My Button',
          },
        );
      });
```
